asyncpg/connection.py

```diff
@@ -311,7 +335,12 @@ async def execute(self, query: str, *args, timeout: float=None) -> str:
         """
         self._check_open()
 
+        # Append to circular buffer of most recent executed statements
```

Comment:
SQLA arg-free statements like BEGIN, COMMIT, etc. end up commit through execute() here and not through prepared statements.